lib/store.js

```diff
@@ -1,24 +1,38 @@
+import { lru } from 'tiny-lru'
+import { DEFAULT_OPTIONS } from './constants.js'
 export class Store {
-  constructor(timeWindowMs) {
+  constructor(
+    timeWindowMs,
+    intervalTimeExpiredKeysEvictionMs,
+    cacheSize = DEFAULT_OPTIONS.cacheSize
+  ) {
     this.timeWindowMs = timeWindowMs
-    this.resetStore()
-    this.interval = setInterval(() => this.resetStore(), timeWindowMs)
-  }
-
-  resetStore() {
-    this.hits = {}
+    this.cache = lru(cacheSize, timeWindowMs)
+    this.interval = setInterval(() => {
+      this.cache.keys().forEach(key => {
+        this.cache.get(key) // it deletes expired keys under the hood and returns undefined
```

Comment:
I've not found this behaviour in the library documentation, so I don't think its safe to use this assumpion

Reply:
`fastify-rate-limit` plugin bases on this behaviour so I assumed that we can take advantage of it as well

Reply:
Additionaly this is the only way to interact with `items` directly and check whether item has expired or not